src/query/models/tags.go

```diff
@@ -454,7 +465,59 @@ func (t Tags) Normalize() Tags {
 	return t
 }
 
-// HashedID returns the hashed ID for the tags.
+// Validate will validate there are tag values, and the
+// tags are ordered and there are no duplicates.
+func (t Tags) Validate() error {
+	n := t.Len()
+	if n == 0 {
+		return errNoTags
+	}
+
+	if t.Opts.IDSchemeType() == TypeGraphite {
+		// Graphite tags are sorted numerically rather than lexically.
+		tags := sortableTagsNumericallyAsc(t)
+		for i := 0; i < n; i++ {
+			if len(tags.Tags[i].Name) == 0 {
+				return fmt.Errorf("tag name empty: index=%d", i)
+			}
+			if i == 0 {
+				continue // Don't check order/unique attributes.
+			}
+
+			if !tags.Less(i-1, i) {
+				return fmt.Errorf("tags out of order: '%s' appears after '%s'",
+					tags.Tags[i-1].Name, tags.Tags[i].Name)
+			}
+			if bytes.Compare(tags.Tags[i-1].Name, tags.Tags[i].Name) == 0 {
+				return fmt.Errorf("tags duplicate: '%s' appears more than once",
+					tags.Tags[i-1].Name)
+			}
+		}
+	} else {
+		// Sorted alphanumerically otherwise.
+		for i := 0; i < n; i++ {
+			if len(t.Tags[i].Name) == 0 {
+				return fmt.Errorf("tag name empty: index=%d", i)
+			}
```

Comment:
Can we do the same for tag value? Prom doesn't allow empty values and it's led to problems for us before

Reply:
Sure thing.